src/libdrakvuf/libdrakvuf.h

```diff
@@ -274,6 +302,32 @@ char *drakvuf_get_current_process_name(drakvuf_t drakvuf,
                                        uint64_t vcpu_id,
                                        x86_registers_t *regs);
 
+
+bool drakvuf_get_current_thread_id( drakvuf_t drakvuf, 
+                                    uint64_t vcpu_id, 
+                                    x86_registers_t *regs,
+                                    uint32_t *thread_id );
+
+bool drakvuf_get_previous_mode( drakvuf_t drakvuf, 
```

Comment:
Great, can you add this link in a comment so that other's can later find it too? Also, what do you think about making this function take the thread address manually instead of restricting it to the current thread only?